### PR TITLE
DPE-2112 - fix: don't raise if user deletion not exist

### PR DESCRIPTION
**Pull request by marcoppenheimer**
Copying over PR from VM charm - https://github.com/canonical/kafka-operator/pull/114

### CODE REVIEW

File: src/provider.py [hunk 1]
```diff
@@ -118,23 +107,28 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         Args:
             event: the event from a related client application needing a user
         """
-        if not self.charm.unit.is_leader():
+        print("1")
```
Comment (deusebio) [inline src/provider.py:110]: **todo** shall we remove this?

File: src/auth.py [hunk 2]
```diff
@@ -331,3 +333,20 @@ def update_user_acls(
         acls_to_remove = current_user_acls - self.new_user_acls
         for acl in acls_to_remove:
             self.remove_acl(**asdict(acl))
+
+    def clear_users(self) -> None:
```
Comment (deusebio) [inline src/auth.py:337]: **question** why is this needed? where is this used?

Comment (marcoppenheimer) [reply src/auth.py:337]: @zmraul - Could you respond here?

Comment (zmraul) [reply src/auth.py:337]: This method is meant to cover for leftover relations. It's meant to be used on update_status or similar, but I wasn't too sure on its usefulness, so it ended up not being used.

**Review by zmraul — APPROVED** [review]

File: src/auth.py (hunk 2 above)
Comment (deusebio) [reply src/auth.py:337]: Oh, I see. But if it is not used, we can probably remove it then, just to have less code to maintain and support

**Review by deusebio — APPROVED** [review]
lgtm
